File: samples/src/main/java/tk/zielony/carbonsamples/WidgetsActivity.java
package tk.zielony.carbonsamples;

import android.os.Bundle;

import java.io.Serializable;

import carbon.component.PaddingItem;
import tk.zielony.carbonsamples.widget.BackdropActivity;
import tk.zielony.carbonsamples.widget.BannerActivity;
import tk.zielony.carbonsamples.widget.BottomNavigationViewActivity;
import tk.zielony.carbonsamples.widget.BottomSheetActivity;
import tk.zielony.carbonsamples.widget.ButtonsActivity;
import tk.zielony.carbonsamples.widget.CheckBoxRadioActivity;
import tk.zielony.carbonsamples.widget.CircularProgressActivity;
import tk.zielony.carbonsamples.widget.DropDownActivity;
import tk.zielony.carbonsamples.widget.ExpandableRecyclerActivity;
import tk.zielony.carbonsamples.widget.ExpansionPanelActivity;
import tk.zielony.carbonsamples.widget.FABActivity;
import tk.zielony.carbonsamples.widget.FlowLayoutActivity;
import tk.zielony.carbonsamples.widget.NavigationViewActivity;
import tk.zielony.carbonsamples.widget.MenusActivity;
import tk.zielony.carbonsamples.widget.ProgressBarsActivity;
import tk.zielony.carbonsamples.widget.RecyclerActivity;
import tk.zielony.carbonsamples.widget.SeekBarActivity;
import tk.zielony.carbonsamples.widget.SnackbarActivity;
import tk.zielony.carbonsamples.widget.TableLayoutActivity;
import tk.zielony.carbonsamples.widget.TabsActivity;
import tk.zielony.carbonsamples.widget.TextFieldsActivity;

@ActivityAnnotation(title = R.string.widgetsActivity_title, layout = R.layout.activity_samplelist)
public class WidgetsActivity extends SampleListActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        initToolbar();

        setItems(new Serializable[]{
                new PaddingItem(getResources().getDimensionPixelSize(R.dimen.carbon_paddingHalf)),
                "Widgets with sample styles, data and applications",
                new SampleActivityItem(BannerActivity.class),
                new SampleActivityItem(CheckBoxRadioActivity.class, R.drawable.carbon_checkbox_checked),
                new SampleActivityItem(ButtonsActivity.class),
                new SampleActivityItem(FABActivity.class, R.drawable.ic_add_circle_black_24dp),
                new SampleActivityItem(CircularProgressActivity.class),
                new SampleActivityItem(MenusActivity.class, R.drawable.ic_menu_black_24dp),
                new SampleActivityItem(ProgressBarsActivity.class),
                new SampleActivityItem(SnackbarActivity.class),
                new SampleActivityItem(TextFieldsActivity.class, R.drawable.ic_text_fields_black_24dp),
                new SampleActivityItem(TabsActivity.class),
                new SampleActivityItem(RecyclerActivity.class, R.drawable.ic_view_stream_black_24dp),
                new SampleActivityItem(ExpandableRecyclerActivity.class),
                new SampleActivityItem(ExpansionPanelActivity.class),
                new SampleActivityItem(DropDownActivity.class, R.drawable.carbon_dropdown),
                new SampleActivityItem(NavigationViewActivity.class, 0, true),
                new SampleActivityItem(SeekBarActivity.class),
                new SampleActivityItem(FlowLayoutActivity.class),
                new SampleActivityItem(TableLayoutActivity.class),
                new SampleActivityItem(BottomNavigationViewActivity.class),
                new SampleActivityItem(BottomSheetActivity.class, 0, true),
                new SampleActivityItem(BackdropActivity.class),
                new PaddingItem(getResources().getDimensionPixelSize(R.dimen.carbon_paddingHalf))
        });
    }

}

File: samples/src/main/java/tk/zielony/carbonsamples/WidgetsActivity.kt
package tk.zielony.carbonsamples

import android.os.Bundle
import tk.zielony.carbonsamples.widget.*

@SampleAnnotation(
        titleId = R.string.widgetsActivity_title,
        layoutId = R.layout.activity_samplelist,
        iconId = R.drawable.ic_widgets_black_24dp
)
class WidgetsActivity : SampleListActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        initToolbar()

        setItems(listOf(
                "Widgets with sample styles, data and applications",
                SampleActivityItem(BannerActivity::class.java),
                SampleActivityItem(CheckBoxRadioActivity::class.java),
                SampleActivityItem(ButtonsActivity::class.java),
                SampleActivityItem(FABActivity::class.java),
                SampleActivityItem(CircularProgressActivity::class.java),
                SampleActivityItem(MenusActivity::class.java),
                SampleActivityItem(ProgressBarsActivity::class.java),
                SampleActivityItem(SnackbarActivity::class.java),
                SampleActivityItem(TextFieldsActivity::class.java),
                SampleActivityItem(TabsActivity::class.java),
                SampleActivityItem(RecyclerActivity::class.java),
                SampleActivityItem(ExpandableRecyclerActivity::class.java),
                SampleActivityItem(ExpansionPanelActivity::class.java),
                SampleActivityItem(DropDownActivity::class.java),
                SampleActivityItem(NavigationViewActivity::class.java, true),
                SampleActivityItem(SeekBarActivity::class.java),
                SampleActivityItem(FlowLayoutActivity::class.java),
                SampleActivityItem(TableLayoutActivity::class.java),
                SampleActivityItem(BottomNavigationViewActivity::class.java),
                SampleActivityItem(BottomSheetActivity::class.java, true),
                SampleActivityItem(BackdropActivity::class.java)
        ))
    }

}
